tests/parametric/test_parametric_endpoints.py

```diff
@@ -0,0 +1,733 @@
+        - span_id: Union[int, str]
+        """
+        with test_library:
+            current_span = test_library.current_span()
+            assert int(current_span.span_id) == 0
+            assert int(current_span.trace_id) == 0
+
+            with test_library.start_span("span_test_current_span") as s1:
+                current_span = test_library.current_span()
+                assert current_span.span_id == s1.span_id
+
+                with test_library.start_span("span_test_current_spans_s2", parent_id=s1.span_id) as s2:
+                    current_span = test_library.current_span()
+                    assert current_span.span_id == s2.span_id
+
+                current_span = test_library.current_span()
+                assert current_span.span_id == s1.span_id
+
+            current_span = test_library.current_span()
+            assert int(current_span.span_id) == 0
+            assert int(current_span.trace_id) == 0
+
+    def test_current_span_from_otel(self, test_agent, test_library):
+        """
+        Validates that /trace/span/current_span can return the Datadog span that was created by the OTEL API.
+
+        Supported Parameters:
+        Supported Return Values:
+        - span_id: Union[int, str]
+        """
+        with test_library:
+            with test_library.otel_start_span("span_test_current_span_from_otel") as s1:
+                current_span = test_library.current_span()
+                assert current_span.span_id == s1.span_id
+                s1.end_span()
+            current_span = test_library.current_span()
+            assert int(current_span.span_id) == 0
+            assert int(current_span.trace_id) == 0
+
+
+@scenarios.parametric
+@features.parametric_endpoint_parity
+class Test_Parametric_DDTrace_Flush:
+    def test_flush(self, test_agent, test_library):
+        """
+        Validates that /trace/span/flush /trace/stats/flush endpoints are implemented and return a successful status code.
+        If these endpoint is not implemented, spans/stats will not be flushed when the test_library contextmanager exits.
+        Traces data may or may not be received by the agent in time for validation. This can introduce flakiness in tests.
+
+        Supported Parameters:
+        Supported Return Values:
+        - trace_flush_status_code: int
+        - stats_flush_status_code: int
+        """
+        # Here we are avoiding using the __exit__() operation on the contextmanager
+        # and instead manually finishing and flushing the span.
+        s1 = test_library.start_span("span").__enter__()
+        s1.finish()
+        assert test_library.flush()
+
+
+@scenarios.parametric
+@features.parametric_endpoint_parity
+class Test_Parametric_DDTrace_Baggage:
+    def test_set_baggage(self, test_agent, test_library):
+        """
+        Validates that /trace/span/set_baggage sets a baggage item on a span.
+
+        Supported Parameters:
+        - span_id: Union[int, str]
```

Comment:
I'm not sure that it makes sense for the API to require a span_id to set baggage, since you can have baggage even without a corresponding span 🤔

Reply:
They only operate on spans. I am just providing documentation/tests for existing behavior. We can update this and refactor the existing tests in a future PR.

cc: @rachelyangdog